src/calibre/devices/kobo/driver.py

```diff
@@ -2587,41 +2610,62 @@ def images_path(self, path, imageId=None):
             path = os.path.join(path, imageId)
         return path
 
-    def _calculate_kobo_cover_size(self, library_size, kobo_size, keep_cover_aspect, is_full_size):
+    def _calculate_kobo_cover_size(self, library_size, kobo_size, expand, keep_cover_aspect, letterbox):
+        # Remember the canvas size
+        canvas_size = kobo_size
+
+        # NOTE: Loosely based on Qt's QSize::scaled implementation
         if keep_cover_aspect:
-            library_aspect = library_size[0] / library_size[1]
-            kobo_aspect = kobo_size[0] / kobo_size[1]
-            if library_aspect > kobo_aspect:
-                kobo_size = (kobo_size[0], int(kobo_size[0] / library_aspect))
+            # NOTE: Py3k wouldn't need explicit casts to return a float
+            # NOTE: Unlike Qt, we round to avoid accumulating errors,
+            #       as ImageOps will then floor via fit_image
+            aspect_ratio = library_size[0] / float(library_size[1])
+            rescaled_width = int(round(kobo_size[1] * aspect_ratio))
```

Comment:
We use `__future__.division` here, so it should not be necessary to convert to a float merely to preserve precision. Also rescaled_width is only used to compare against other numbers for truthiness, so it shouldn't be necessary to use both int() and round().

Reply:
Good point, I'd missed that, which explains why no-one ever complained that the AR handling was broken, and why I was seeing different results when I tested that logic in a Python shell ;). I'll kill the casts, thanks.

`rescaled_width` *can* be returned as, well, the width in some circumstances, and round() returns a float, so, AFAICT, it does need both.